jdbc/src/main/java/org/apache/iotdb/jdbc/IoTDBQueryResultSet.java

```diff
@@ -716,10 +689,75 @@ public boolean next() throws SQLException {
     if (emptyResultSet) {
       return false;
     }
-    record = Utils.getRowRecord(byteBuffer, columnTypeDeduplicatedList);
+    constructOneRow();
     return true;
   }
 
+  private void constructOneRow() {
+    record[0] = new byte[Long.BYTES];
+    tsQueryDataSet.time.get(record[0]);
+    for (int i = 0; i < tsQueryDataSet.bitmapList.size(); i++) {
+      ByteBuffer bitmapBuffer = tsQueryDataSet.bitmapList.get(i);
+      // another new 8 row, should move the bitmap buffer position to next byte
+      if (rowsIndex % 8 == 0) {
+        bitmapBuffer.get();
+      }
+
+      record[i+1] = null;
+      // the column value is not null, we should move the corresponding value buffer to next position
+      if (!isNull(i, rowsIndex)) {
+        ByteBuffer valueBuffer = tsQueryDataSet.valueList.get(i);
+        TSDataType dataType = TSDataType.valueOf(columnTypeList.get(i));
+        switch (dataType) {
+          case BOOLEAN:
+            record[i+1] = new byte[1];
+            valueBuffer.get(record[i+1]);
+            break;
+          case INT32:
+            record[i+1] = new byte[Integer.BYTES];
+            valueBuffer.get(record[i+1]);
+            break;
+          case INT64:
+            record[i+1] = new byte[Long.BYTES];
+            valueBuffer.get(record[i+1]);
+            break;
+          case FLOAT:
+            record[i+1] = new byte[Float.BYTES];
+            valueBuffer.get(record[i+1]);
+            break;
+          case DOUBLE:
+            record[i+1] = new byte[Double.BYTES];
```

Comment:
also, all non-text fields could be treated as the same with the timestamp column

Reply:
what if the record value has been set to null before